From 1fc205e6d69e9fed48488bb5a64c721c09abeffb Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sat, 21 Jan 2017 11:09:45 +1100
Subject: [PATCH 44/47] dpnet: Return S_OK from IDirectPlay8Peer ReturnBuffer

When DPNSUCCESS_PENDING is received from the message handler
we should wait until the application has finished with it.
eg. Called ReturnBuffer
---
 dlls/dpnet/client.c | 5 +++++
 dlls/dpnet/peer.c   | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/dlls/dpnet/client.c b/dlls/dpnet/client.c
index 31e0c56..b749e92 100644
--- a/dlls/dpnet/client.c
+++ b/dlls/dpnet/client.c
@@ -1205,6 +1205,11 @@ static HRESULT dplay_connected(SOCKET sock, void *serveraddr, size_t addrsize, D
                                     received.dwReceiveFlags     = 0;
 
                                     hr = msghandler(NULL, DPN_MSGID_RECEIVE, &received);
+                                    if(hr == DPNSUCCESS_PENDING)
+                                    {
+                                        TRACE("Pending Returned\n");
+                                        hr = S_OK;
+                                    }
                                     if(hr != S_OK)
                                         ERR("Message Handle for DPNMSG_RECEIVE failed (0x%08x).\n", hr);
 
diff --git a/dlls/dpnet/peer.c b/dlls/dpnet/peer.c
index 52bd6c9..16f19a8 100644
--- a/dlls/dpnet/peer.c
+++ b/dlls/dpnet/peer.c
@@ -716,7 +716,7 @@ static HRESULT WINAPI IDirectPlay8PeerImpl_ReturnBuffer(IDirectPlay8Peer *iface,
 {
     FIXME("(%p)->(%x,%x): stub\n", iface, hBufferHandle, dwFlags);
 
-    return DPNERR_GENERIC;
+    return S_OK;
 }
 
 static HRESULT WINAPI IDirectPlay8PeerImpl_GetPlayerContext(IDirectPlay8Peer *iface, const DPNID dpnid,
-- 
1.9.1

